From 26c63732c48828ad9d052e3dc0b44cd86d23b11e Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Mon, 1 Feb 2016 16:10:44 +0800
Subject: [PATCH 0920/5242] MLK-12370 ARM: dts: imx7d-12x12-lpddr3-arm2:
 disable sim1 node in flexcan dts file

commit  c488ac352f83527856c477321b79babbc59101dc from
https://source.codeaurora.org/external/imx/linux-imx.git

The sim1 has pin conflicts with flexcan1,flexcan2 and sai1.
By default, imx7d 12x12 lpddr3 arm2 default dts enable sim1 node
and disable flexcan1, flexcan2, sai1 nodes.

The patch do two things:
- disable sim1 node in extended flexcan dts file
- remove redundant sim1 node in default dts file

Signed-off-by: Fugang Duan <B38611@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/imx7d-12x12-lpddr3-arm2-flexcan.dts   |    4 ++++
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts      |    8 +-------
 2 files changed, 5 insertions(+), 7 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-flexcan.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-flexcan.dts
index 1888c87..90ea885 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-flexcan.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-flexcan.dts
@@ -23,3 +23,7 @@
 &sai1 {
 	status = "disabled";
 };
+
+&sim1 {
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
index b78b0cd..f991a00 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
@@ -905,7 +905,7 @@
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&pinctrl_sim1_1>;
 	pinctrl-1 = <&pinctrl_sim1_1>;
-	status = "disabled";
+	status = "okay";
 };
 
 &uart1 {
@@ -989,12 +989,6 @@
 	status = "disabled";
 };
 
-&sim1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_sim1_1>;
-	status = "okay";
-};
-
 &wdog1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_wdog>;
-- 
1.7.9.5

